### DISCUSSION TITLE
github and notitifications

### DISCUSSION THREAD

**Discussion by stefanw1337** [community/community#144934] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

So I'm trying to get notified on an update for a project. Sounds simple enough? But I don't understand github and why everything needs to be so complicated, let that be the underline here. And I've been trying for years to understand its complexity.

So this project has a nightly build, which as you know, makes a new file every night. But I don't want to download it if it doesn't bring anything NEW. So I need to have it check for new commits on the master-branch here.
https://github.com/<user>/<project>/commits/master/

But right now it doesn't, I get notified of merged commits into <user>:missions (for instance), which probably is an sub-branch, or something.

So while https://github.com/<user>/<project>/commits/master/ shows for instance: "Upgrade to .NET 8" the notification says something else.

And my notifications, I've tried different things every time:
user:<user> head:master status:success
(and I understood stable is the "stable" release, and "master" is the nightly/unstable version for "head:").

But as I said this seem to take sub-branch as well, or something. I don't know.

And so I tried googling it:
https://stackoverflow.com/questions/9845655/how-do-i-get-notifications-for-commits-to-a-github-repository

And it says:
`create a pull request for yourself from the selected branch, e.g. from master of head project to master of your fork:

    base repository: your/project ; base: master <-- head repository: original/project ; compare: master
`

Is this the sentence I should put in the notification? I tried, and it keeps showing the same as before.. Seems to work like a filter, so when I press it, it grabs whatever the filter is?

So I kinda give up right now, I need someone to help me on this. 
And no, I don't want it in RSS. I'd like to have email or push notification (for mobile, android). I have put it in RSS reader, and it works. But then I have to set it up to mail me, and I haven't bothered with it. Help.

**Comment by Amrit1604** [comment, marked as answer]
To get notified only when there’s a new commit on the master branch, try the following methods:

Option 1: GitHub Notifications (Filtered)
Watch the repository, but select "Custom" instead of "All activity."
Enable "Pushes" and uncheck "Pull Requests". This reduces noise from merges and PRs from other branches.
Option 2: GitHub Actions (Custom Notification)
If you have access to the repository, create a GitHub Action that triggers only on master branch commits:

```
name: Notify on Master Commit

on:
  push:
    branches:
      - master

jobs:
  notify:
    runs-on: ubuntu-latest
    steps:
      - name: Send Notification
        run: curl -X POST -H "Content-Type: application/json" \
          -d '{"text": "New commit on master branch"}' \
          YOUR_WEBHOOK_URL


```



Replace YOUR_WEBHOOK_URL with a service like Slack, Discord, or Pushover for email or push notifications.

Option 3: Use IFTTT for Push Notifications
Set up an IFTTT applet with:
Trigger: New commit on master branch (use GitHub’s feed URL for master).
Action: Send a push notification or email.
These options should help you get notifications only for new commits on master, without the extra noise. Let me know if you need help setting any of these up!

**Reply by stefanw1337** [reply]
Thank you so much. I will try these options!

I think the first one might solve the issue with sub-branches. But the problem was also that the actual commit; I wasn't notified with the "Upgrade to .NET 8" at all. Just some other sub-branch stuff, which means it didn't work as I wanted. The RSS does show them, and that works. I could perhaps download some kind of addon to my browser that will give sound and maybe marquee the new update over the screen or something as an RSS option.

Option 2, do I have to do anything else if I replace web_hook_url with "Discord". Does it know what server, or does it pop up in Discord as an available hook so I can set it there?

**Reply by stefanw1337** [reply]
Also I can only find on "Watch":
Issues
Pull requests
Releases
Discussions
(Discussions are not enabled for this repository)
Security alerts

So unchecked pull request there, but I can't see any pushes, unless that's "Releases"? But is that stable release as well, or just stable release?
Edit: Seems to be both, since I can change branch, and Watch remains the same.

Found Feedbro, RSS addon for browser, but for sound and notification it requires a json-file.. Why so complicated, bro? I can't program json.

I made AI make the rule-file for me, let's see if it works.
